jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/NativeImagePathResourceFactory.java

```diff
@@ -0,0 +1,71 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+/**
+ * GraalVM Native-Image {@link PathResourceFactory}.
+ */
+public class NativeImagePathResourceFactory extends PathResourceFactory
+{
+    static final boolean ENABLE_GRAALVM_RESOURCE_SCHEME = (System.getProperty("org.graalvm.nativeimage.kind") != null);
+
+    public NativeImagePathResourceFactory()
+    {
+        if (ENABLE_GRAALVM_RESOURCE_SCHEME)
+        {
+            initNativeImageResourceFileSystem();
+        }
+    }
+
+    private void initNativeImageResourceFileSystem()
+    {
+        try
+        {
+            URI uri = new URI("resource:/");
+            try
+            {
+                Path.of(uri);
+            }
+            catch (FileSystemNotFoundException e)
+            {
+                FileSystems.newFileSystem(uri, Collections.emptyMap());
+            }
+        }
+        catch (IOException | URISyntaxException | RuntimeException ignore)
+        {
+            // ignore
```

Comment:
Shouldn't we log exceptions as warnings here, as they may be thrown by `newFileSystem` and that feels significant?

Reply:
I don't think so, since there's nothing we can do here.

If any of this fails, subsequent attempts to `Path.of`/`Paths.get` a `resource:` URI will throw an exception from within the calling stack.

We try-catch `RuntimeException` for example to ignore a race condition between us trying to initialize the filesystem and application code that triggers the filesystem initialization independently (we might get a `FileSystemAlreadyExistsException` just at an inopportune time);

Reply:
I think at least a debug log would be helpful.   If somebody is getting URI exceptions elsewhere because the filesystem didn't mount, it would be good to be able to see the exceptions without needing to modify code